File: rot13.py
"""
Write a function to decrypt an ROT13 Enconded message.

message = "Ohg, bssvpre, V qvqa'g pngpu gurfr -- gurl ner zl crg svfu naq V whfg\
 oevat gurz urer gb fjvz. Jura gurl'er qbar gurl whzc onpx vagb gur ohpxrg."
"""

# Write your code here.

import string


def decoder():
    message = "Ohg, bssvpre, V qvqa'g pngpu gurfr -- gurl ner zl crg svfu naq V whfg\
 oevat gurz urer gb fjvz. Jura gurl'er qbar gurl whzc onpx vagb gur ohpxrg.".lower()

    punct = string.punctuation
    space = string.whitespace

    decoded = ''
    for letters in message:
        if letters in punct or letters in space:
            decoded += letters
        else:
            if ord(letters) >= 110:
                correct = ord(letters) - 13
            elif ord(letters) <= 109:
                correct = ord(letters) + 26 - 13
            back_to_char = chr(correct)
            decoded += back_to_char

    return decoded


decoder()

File: rot13.js
/*
  Created by Steve Hanlon on 5/30/17.
 */

/*
Write a JavaScript program to decrypt an ROT13 Encoded message.

message = "Ohg, bssvpre, V qvqa'g pngpu gurfr -- gurl ner zl crg svfu naq V whfg\
 oevat gurz urer gb fjvz. Jura gurl'er qbar gurl whzc onpx vagb gur ohpxrg."
*/


var message = "Ohg, bssvpre, V qvqa'g pngpu gurfr -- gurl ner zl crg svfu naq V whfg\
 oevat gurz urer gb fjvz. Jura gurl'er qbar gurl whzc onpx vagb gur ohpxrg.";

var encoded = message.toLowerCase();
var alphabet = "abcdefghijklmnopqrstuvwxyz, '-.";
var cesar13 =  "nopqrstuvwxyzabcdefghijklm, '-.";

var decoded = [];

function cipher(msg) {
    for (var i = 0; i < encoded.length; i += 1) {
        var enchar = encoded[i];
        var encloc = cesar13.indexOf(enchar);
        var decodedChar = alphabet[encloc];
        decoded.push(decodedChar);
    }
    var result = decoded.join('');
    document.write(result);
}


cipher(message);